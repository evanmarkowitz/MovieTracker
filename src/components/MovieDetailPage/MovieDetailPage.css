.backdrop {
  width: 100%;
}

.poster{
  margin-top: 30vh;
  margin-left: 45px;
}

#movie-detail-page {
  text-align: left;
}

.movie-info {
  margin-left: 45px;
  display: flex;
  flex-direction: column;
}

h2{
  /* width: 100%; */
  margin: 0;
  align-self: center;
}

h3 {
  margin: 0;
}

#movie-title {
  background-color: transparent;
  color: #333333;
  width: 100%;
  text-align: center;
  text-shadow: 2px 2px lightgrey;
  font-size: 30px;
}

.synopsis {
  width: 80%;
}