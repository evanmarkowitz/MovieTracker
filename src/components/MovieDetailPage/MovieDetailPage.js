import React from 'react';
import { connect } from 'react-redux';
import './MovieDetailPage.css';
import Overdrive from 'react-overdrive'
import { createRequireFromPath } from 'module';
import { RSA_NO_PADDING } from 'constants';

const MovieDetailPage = ({ title, releaseDate, synopsis, rating, voteCount, poster, id, key, backdrop}) => {
    const backDropUrl = 'http://image.tmdb.org/t/p/w1280'
    const styling = {
      backgroundImage: `url(${backDropUrl}${backdrop})`,
      backgroundSize: 'cover',
      height: '55vh',
    }
    return (
        <div className="movieDetail" style={styling} id='movie-detail-page'>
         
            <Overdrive id={id}>
              <img src={`http://image.tmdb.org/t/p/w154${poster}`} alt={`A movie poster for ${title}.`} className='poster'/>
            </Overdrive>
            <section className='movie-info'>
            <h2 id='movie-title'>{title.toUpperCase()}</h2>
            {/* <p>Release: {genres}</p> */}
            <p>This movie is a {rating} out of 10 based on {voteCount} people's opinions.</p>
            <h3>Synopsis:</h3> 
            <p className ='synopsis'>{synopsis}</p>
            </section>
        </div>
    )
}

const mapStateToProps = (state) => ({
    movies: state.movies
  })
  
  export default connect(mapStateToProps)(MovieDetailPage)