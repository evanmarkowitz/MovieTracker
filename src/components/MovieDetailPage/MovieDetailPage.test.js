import React from 'react';
import MovieDetailPage from './MovieDetailPage';
import { shallow } from 'enzyme';
import configureStore from 'redux-mock-store';


describe('MovieDetailPage', () => {
  let wrapper;
  const initialState = [];
  let store
  const mockStore = configureStore();

  beforeEach(() => {
    store = mockStore(initialState);
    wrapper = shallow(<MovieDetailPage store={store}/>)
  })

  it('should match the snapshot', () => {
    expect(wrapper).toMatchSnapshot()
  })
})