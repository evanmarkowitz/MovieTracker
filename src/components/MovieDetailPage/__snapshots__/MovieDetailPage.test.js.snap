// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MovieDetailPage MovieDetail Snapshot should match the snapshot 1`] = `ShallowWrapper {}`;
